modules/ROOT/pages/migration/index.adoc

```diff
@@ -1,729 +1,1008 @@
+        password: "guest",
+    },
+});
+----
+a|
+[source, javascript, indent=0]
+----
+const subscriptionsEngine = new Neo4jGraphQLAMQPSubscriptionsEngine({
+    connection: {
+        hostname: "localhost",
+        username: "guest",
+        password: "guest",
+    },
+});
+----
+|===
 
-|`averageRating_gt: Float`
-|`averageRating_GT: Float`
+=== Custom subscription plugins
 
-|`averageRating_gte: Float`
-|`averageRating_GTE: Float`
-|===
+The underlying subscription system has not changed.
+Custom behavior can be implemented the same way, by creating a class implementing the interface described in xref::subscriptions/engines.adoc#custom-subscription[Subscriptions engines].
 
-For filtering on relationship fields, the `_some`, `_none`, `_single` and `_every` filters are not yet implemented.
+However, if using TypeScript, the exported interface to implement these classes has been renamed from `Neo4jGraphQLSubscriptionsPlugin` to `Neo4jGraphQLSubscriptionsEngine`.
 
-[[migration-guide-queries-options]]
-=== Sorting and Pagination (`options`)
+== Updated directives
 
-==== Sorting
+A number of directives and their arguments have been renamed in order to make using `@neo4j/graphql` more intuitive.
+Here is a table with all the changes:
 
-Sorting has changed somewhat in `@neo4j/graphql`. For the example being used in this page, the `_MovieOrdering` type in `neo4j-graphql-js` was an enum which looked like the following:
+[cols="1,2,2"]
+|===
+|Before | Now | Example
 
+|`@alias`
+|Properties in the alias directive are now automatically escaped using backticks. 
+If you were using backticks in the `property` argument of your `@alias` directives, you should now remove the escape strings as this is covered by the library.
+a|
 [source, graphql, indent=0]
 ----
-enum _MovieOrdering {
-    title_asc
-    title_desc
-    averageRating_asc
-    averageRating_desc
+type User {
+    id: ID! @id
+    username: String! @alias(property: "dbUserName")
 }
 ----
 
-You could then query all movies ordered by title ascending by executing:
-
+|`@callback`
+|Renamed to `@populatedBy`.
+Additionally, the `name` argument has been renamed to `callback` and it is still used to specify the callback used to populate the field's value.
+a|
+.Before
 [source, graphql, indent=0]
 ----
-query {
-    Movie(orderBy: [title_asc]) {
-        title
-    }
+type User {
+  id: ID! @callback(name: "nanoid", operations: [CREATE])
+  firstName: String!
+  surname: String!
 }
 ----
 
-In `@neo4j/graphql`, the sorting type `MovieSort` has become an input type with each field as an enum, like follows:
-
+.Now
 [source, graphql, indent=0]
 ----
-enum SortDirection {
-    ASC
-    DESC
-}
-
-input MovieSort {
-    title: SortDirection
-    averageRating: SortDirection
+type User {
+  id: ID! @populatedBy(callback: "nanoid", operations: [CREATE])
+  firstName: String!
+  surname: String!
 }
 ----
 
-To fetch all movies sorted by title ascending as per above, you would execute:
+|`nanoid`
```

Comment:
We don't have a directive named `nanoid` this change is a follow-up of the previous one `callback` to `populatedBy`, in the same way, it's changed the directive name, it's changed also the configuration option name!
The real change here is this:
```typescript
new Neo4jGraphQL({
  typeDefs,
  features: { // changed from config
    populatedBy: { // changed from callback
      callbacks: {
        nanoid: () => { return nanoid(); }
      }
    }
  }
});
```